internal/client/ssh.go

```diff
@@ -0,0 +1,386 @@
+	if c.connHasRoot {
+		sudoPrefix = "sudo"
+	}
+	cmd := fmt.Sprintf("%s rm %s", sudoPrefix, BlueChiControllerConfdDirectory+file)
+	output, err := session.Output(cmd)
+	if err != nil {
+		return fmt.Errorf("failed to remove controller config file: %s", string(output))
+	}
+
+	return nil
+}
+
+func (c *SSHClient) RestartBlueChiController() error {
+	session, err := c.newSSHSession()
+	if err != nil {
+		return err
+	}
+	defer session.Close()
+
+	sudoPrefix := ""
+	if c.connHasRoot {
+		sudoPrefix = "sudo"
+	}
+	output, err := session.Output(fmt.Sprintf("%ssystemctl start bluechi-controller", sudoPrefix))
+	if err != nil {
+		return fmt.Errorf("failed to restart controller service: %s", string(output))
+	}
+
+	return nil
+}
+
+func (c *SSHClient) StopBlueChiController() error {
+	session, err := c.newSSHSession()
+	if err != nil {
+		return err
+	}
+	defer session.Close()
+
+	sudoPrefix := ""
+	if c.connHasRoot {
+		sudoPrefix = "sudo"
+	}
+	output, err := session.Output(fmt.Sprintf("%s systemctl stop bluechi-controller", sudoPrefix))
+	if err != nil {
+		return fmt.Errorf("failed to stop controller service: %s", string(output))
+	}
+
+	return nil
+}
+
+func (c *SSHClient) CreateAgentConfig(file string, cfg BlueChiAgentConfig) error {
+	session, err := c.newSSHSession()
+	if err != nil {
+		return err
+	}
+	defer session.Close()
+
+	sudoPrefix := ""
+	if c.connHasRoot {
+		sudoPrefix = "sudo"
+	}
+	cmd := fmt.Sprintf("%s bash -c 'echo \"%s\" > %s'", sudoPrefix, cfg.Serialize(), BlueChiAgentConfdDirectory+file)
+	output, err := session.Output(cmd)
+	if err != nil {
+		return fmt.Errorf("failed to create agent config file: %s", string(output))
+	}
+
+	return nil
+}
+
+func (c *SSHClient) RemoveAgentConfig(file string) error {
+	session, err := c.newSSHSession()
+	if err != nil {
+		return err
+	}
+	defer session.Close()
+
+	sudoPrefix := ""
+	if c.connHasRoot {
+		sudoPrefix = "sudo"
+	}
+	cmd := fmt.Sprintf("%s rm %s", sudoPrefix, BlueChiAgentConfdDirectory+file)
+	output, err := session.Output(cmd)
+	if err != nil {
+		return fmt.Errorf("failed to remove agent config file: %s", string(output))
+	}
+
+	return nil
+}
+
+func (c *SSHClient) RestartBlueChiAgent() error {
+	session, err := c.newSSHSession()
+	if err != nil {
+		return err
+	}
+	defer session.Close()
+
+	sudoPrefix := ""
+	if c.connHasRoot {
+		sudoPrefix = "sudo"
+	}
+	output, err := session.Output(fmt.Sprintf("%s systemctl start bluechi-agent", sudoPrefix))
+	if err != nil {
+		return fmt.Errorf("failed to restart agent service: %s", string(output))
+	}
+
+	return nil
+}
+
+func (c *SSHClient) StopBlueChiAgent() error {
+	session, err := c.newSSHSession()
+	if err != nil {
+		return err
+	}
+	defer session.Close()
+
+	sudoPrefix := ""
+	if c.connHasRoot {
+		sudoPrefix = "sudo"
+	}
+	output, err := session.Output(fmt.Sprintf("%s systemctl stop bluechi-agent", sudoPrefix))
+	if err != nil {
+		return fmt.Errorf("failed to stop agent service: %s", string(output))
+	}
+
+	return nil
+}
```

Comment:
All these single SSH command methods should be consolidated into a single method that gets the command as a parameter. The command should not include `sudo` and this single method can prepend if needed.

Reply:
Applied your suggested change. Looks cleaner now. Thanks!